src/tests/rules/prefer-repository-shorthand.test.ts

```diff
@@ -0,0 +1,124 @@
+import rule from "../../rules/prefer-repository-shorthand.js";
+import { ruleTester } from "./ruleTester.js";
+
+ruleTester.run("prefer-repository-shorthand", rule, {
+	invalid: [
+		{
+			code: `{
+			"repository": {
+				"type": "git",
+				"url": "https://github.com/JoshuaKGoldberg/create-typescript-app"
+			}
+		}`,
+			errors: [
+				{
+					message:
+						"Use shorthand repository URL for GitHub repository",
```

Comment:
Fun fact, tests can go off `messageId` too! IMO this is preferable. Especially when the messages get complex.

```suggestion
					messageId: "useShorthand",
```